### ISSUE TITLE
[question] How to prevent re-running build when creating multiple packages from one prebuilt zip

### ISSUE THREAD

**Issue by edahlseng** [conan-io/conan#9479]
I'm currently working with an SDK for cross-compiling onto an embedded target, and am looking to package this SDK into a Conan package. This SDK is available via a zip file, and is quite large. I'm trying to package it up into a generic `sdk` package, and a `toolchain` package, so that I can include them in the `requires` and `build_requires` sections, respectively.

Since the zip file is relatively large, I've tried to utilize the `build_id` method to prevent downloading the file multiple times when producing these packages. Unfortunately, this doesn't appear to be working as expected.

The `conanfile.py` that I'm working with looks roughly like so:

```python
from conans import ConanFile, tools

class Sdk(ConanFile):
    name = "sdk"
    version = "1.0.0"
    settings = "os"
    options = { "toolchain": [True, False] }

    def layout(self):
        self.folders.build = "build"

    def build_id(self):
        del self.info_build.options.toolchain

    def build(self):
        # Download and unpack zip

    def set_name(self):
        self.name = self.name + ("-toolchain" if self.options.toolchain else "")

    def package(self):
        if self.options.toolchain:
            # Package toolchain files
        else:
            # Package SDK files

```

I'm running the following:

```shell
conan create . -s os=Macos -o toolchain=False
conan create . -s os=Macos -o toolchain=True
```

I expected that the second command could skip over the build step, but unfortunately both commands run the build step. Even if I run the first command twice in a row, the build step is executed again, so I feel like I must be misunderstanding something. How can I share the downloaded files across these multiple package options?

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @edahlseng 

Yes, the ``build_id`` feature only works for packages with the same name. Due to the ``set_name`` implementation it seems that you are creating different packages with different names. Furthermore, ``options`` are not evaluated at ``set_name()`` time, I am not sure how this is working...)

The idea of ``build_id`` is that you can reuse a build of the same package that output different binaries (for example shared/static library, or Debug/Release), and then package in an individual binary the different artifacts. But still they should be an output of the same build, same source code, not completely different things for different names.

For your use case, the only thing that I could imagine of is to stablish some convention in the ``build()`` method to use an existing pre-defined system folder, and reuse it if already existing, but seems quite a hack, I guess it could have other problems, I am not sure if the efficiency gain is worth it. The thing is that packages like these SDKs are typically built once, not that they are rebuilt every day, so "wasting" 2 hours to build it instead of 1, if done 1 per month, maybe is not a big deal?

**Comment by edahlseng** [comment]
Hi @memsharded, thank you for the response.

Indeed, I see that `options` are not evaluated at `set_name()` time. I had included that in my example above to show my final intent, but had been testing without it as I tried to figure out what was going on. It seems that the build was still re-running even when I used the same package name.

Based on your suggestion I've split the conanfile into two separate files, and found a way to cache the download (by far the slowest part). There's still a duplicated delay related to unzipping and packaging everything, but it's workable. Thanks for your help!
